pcdsutils/info.py

```diff
@@ -0,0 +1,471 @@
+
+
+@dataclasses.dataclass
+class Experiment(_LogbookInfo):
+    """Experiment information."""
+    #: The experiment name.
+    name: str = ""
+    #: The experiment description.
+    description: str = ""
+    #: The experiment start time.
+    start_time: str = ""
+    #: The experiment end time.
+    end_time: str = ""
+    #: The lead account information.
+    leader_account: str = ""
+    #: The lead contact information.
+    contact_info: str = ""
+    #: The posix group of the experiment.
+    posix_group: str = ""
+    #: Experiment parameter information.
+    params: Dict[str, str] = dataclasses.field(default_factory=dict)
+
+    @classmethod
+    def from_logbook(
+        cls, instrument: Instrument, experiment_name: Optional[str] = None
+    ) -> Experiment:
+        """
+        Get Experiment information given an instrument/experiment name.
+
+        Parameters
+        ----------
+        instrument : Instrument
+            The instrument instance.
+
+        experiment_name : str, optional
+            Experiment name. Defaults to the active experiment.
+        """
+        # TODO: how do we get information about a given experiment?
+        try:
+            resp = requests.get(
+                WS_URL + "/lgbk/ws/activeexperiment_for_instrument_station",
+                dict(
+                    instrument_name=instrument.hutch,
+                    station=instrument.station
+                ),
+            ).json()
+        except Exception:
+            return cls()
+
+        return cls._response_to_instance(resp)
+
+
+@dataclasses.dataclass
+class RunFiles(_LogbookInfo):
+    """For a given run, information about the files generated."""
+    files: List[str] = dataclasses.field(default_factory=list)
+    num_files: int = 0
+
+    @classmethod
+    def from_logbook(cls, experiment: Experiment, run: Run) -> RunFiles:
+        """
+        Get RunFile information given an experiment/run, queried from the
+        logbook.
+
+        Parameters
+        ----------
+        experiment : Experiment
+            The experiment instance.
+
+        run : Run
+            The Run instance.
+        """
+        try:
+            resp = requests.get(
+                f"{WS_URL}/lgbk/{experiment.name}"
+                f"/ws/{run.num}/files_for_live_mode"
+            )
+            info = resp.json()
+        except Exception:
+            return RunFiles(valid=False)
+
+        valid = info.get("success", False)
+        files = [DATA_PATH + fn.lstrip("/") for fn in info.get("value", [])]
+        run_files = RunFiles(valid=valid, files=files)
+        run_files.num_files = cls._get_num_files(run_files.files)
+        return run_files
+
+    @staticmethod
+    def _get_num_files(files: List[str]) -> int:
+        """Get the number of files in this run.  Some are filtered."""
+        num_files = 0
+        for fn in files:
+            # Note: I don't know why this is
+            if "c00" in fn and "-s8" not in fn:
```

Comment:
s0x are the nominal data streams (for LCLS1). They start counting at 0, so you already have s00, up to however many dss-nodes you used.  s8x would be IOC recorders.
c0x are chunks: we start writing the c00-files and when the file gets too big, it will be closed and a new file will be opened. This is done by the DAQ stopping triggers for a bit & then restarting them. This tripped up a scan at some point if I remember right. With the jungfrau4M, CXI is the hutch mostly likely to chunk and MEC rarely ever does.